### PR TITLE
Gotowy slider.

### PR DESCRIPTION
**Pull request by damianc**
Działający skrypt slidera.
Ticket #6235
https://bugzilla.stxnext.pl/show_bug.cgi?id=6235

### CODE REVIEW

File: css/slider.css [hunk 1]
```diff
@@ -0,0 +1,80 @@
+.slideshowDiv {
+	border: solid 1px #aaa;
```
Comment (Gruszks) [inline css/slider.css]: Always use 4 spaces.

File: index.html [hunk 2]
```diff
@@ -0,0 +1,16 @@
+<!doctype html>
+<html>
+  <head>
+    <meta charset="utf-8" />
+    <title>Slider</title>
+    <link rel="stylesheet" href="css/slider.css" />
```
Comment (Gruszks) [inline index.html]: 4 spaces.

File: js/slider-lib.js [hunk 3]
```diff
@@ -0,0 +1,144 @@
+var Slider = function () {
```
Comment (Gruszks) [inline js/slider-lib.js]: Make this module as a jquery plugin.

File: js/slider.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+(function () {
+
+	var slider = new Slider();
+    var sliderContainer = document.querySelector( '#slideshow' );
+    var paths = [ 'sample-slides/desert.jpg', 'sample-slides/field.jpg', 'sample-slides/forest.jpg' ];
```
Comment (Gruszks) [inline js/slider.js]: This is wrong images should be in HTML structure.

File: slider.css [hunk 5]
```diff
@@ -0,0 +1,9 @@
+.slideshowDiv {
```
Comment (Gruszks) [inline slider.css]: This file is needed?

**Comment by damianc** [comment]
Ticket #6235
https://bugzilla.stxnext.pl/show_bug.cgi?id=6235

Przepisanie slidera na postać wtyczki jQuery.

**Comment by Gruszks** [comment]
This branch has conflicts that must be resolved

File: index.html [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+    </head>
+    <body>
+
+        <ul id="views-slides">
+            <li> <a href="#"> <img src="sample-slides/desert.jpg" /> </a> </li>
+            <li> <a href="#"> <img src="sample-slides/field.jpg" /> </a> </li>
+            <li> <a href="#"> <img src="sample-slides/forest.jpg" /> </a> </li>
+        </ul>
+
+        <script src="http://code.jquery.com/jquery-1.11.3.min.js"></script>
```
Comment (Gruszks) [inline index.html]: Download the library and add it to the project.

File: index.html [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+        <script src="js/slider.jquery.js"></script>
+        <script>
+            $( '#views-slides' ).slider();
```
Comment (Gruszks) [inline index.html]: Where is document ready? Move it to separate file like app.js

File: js/slider.jquery.js [hunk 8]
```diff
@@ -0,0 +1,99 @@
+(function ( $ ) {
+
```
Comment (Gruszks) [inline js/slider.jquery.js]: Remove while line.

File: js/slider.jquery.js [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+        var opts = $.extend({
+            runWith: 0,
+            activeSlide: 0
+        }, options );
+
+
+        /*
+         * Wrapping exisiting listed slides into the plugin's layer.
+         */
+        var slideshowDiv = $( '<div/>' ).addClass( 'slideshow' );
```
Comment (Gruszks) [inline js/slider.jquery.js]: Never space after ( and ). Fix them all.

File: js/slider.jquery.js [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+        var that = $( this );
```
Comment (Gruszks) [inline js/slider.jquery.js]: Var declarations always at the top of function.

File: js/slider.jquery.js [hunk 11]
```diff
@@ -0,0 +1,99 @@
...
+        var nextSlideButton = $( '<div/>' ).addClass( 'slideButton nextSlideButton' );
+        $( this ).parent().append( prevSlideButton ).append( nextSlideButton );
+
+
+        /*
+         * Rendering active slide indicators.
+         */
+        var images = $( this ).find( 'li a img' ).length;
+        var indicatorsDiv = $( '<div/>' ).addClass( 'indicatorsDiv' );
+        for ( var i = 0; i < images; i++ ) {
```
Comment (Gruszks) [inline js/slider.jquery.js]: Use jQuery each or install underscore.

**Comment by damianc** [comment]
6235
https://bugzilla.stxnext.pl/show_bug.cgi?id=6235

File: index.html [hunk 12]
```diff
@@ -0,0 +1,21 @@
...
+<html>
+    <head>
+        <meta charset="utf-8" />
+        <title>Slider</title>
+        <link rel="stylesheet" href="css/slider.jquery.css" />
+    </head>
+    <body>
+
+        <ul id="views-slides">
+            <li> <a href="#"> <img src="sample-slides/desert.jpg" /> </a> </li>
```
Comment (Gruszks) [inline index.html]: LI
    a
         img

File: js/slider.jquery.js [hunk 13]
```diff
@@ -0,0 +1,99 @@
+(function ($) {
+
+    $.fn.slider = function (options) {
+
+
+        return this.each( function () {
+
+            var opts;
```
Comment (Gruszks) [inline js/slider.jquery.js]:
```
            var opts,
                  slideshowDiv,
                  that,
                  slideTo,
                  prevSlideButton,
                  nextSlideButton,
                  images,
                  indicatorsDiv,
                  indicator;

```

File: js/slider.jquery.js [hunk 14]
```diff
@@ -0,0 +1,99 @@
...
+            var prevSlideButton;
+            var nextSlideButton;
+            var images;
+            var indicatorsDiv;
+            var indicator;
+
+            /*
+             * Options of the slideshow.
+             */
+            opts = $.extend({
```
Comment (Gruszks) [inline js/slider.jquery.js]: use options

File: js/slider.jquery.js [hunk 15]
```diff
@@ -0,0 +1,99 @@
...
+            }
+            $(this).parent().append(indicatorsDiv);
+
+
+            /*
+             * Attaching events to previous and next slide buttons.
+             */
+            $('.slideButton').on('click', function () {
+
+                var button = $(this);
```
Comment (Gruszks) [inline js/slider.jquery.js]: ditto with variables

File: js/slider.jquery.js [hunk 16]
```diff
@@ -0,0 +1,99 @@
...
+                var activeSlide = opts.activeSlide;
+                var nextOne;
+
+                /* Previous and next slide button event. */
+                if (button.hasClass('prevSlideButton')) nextOne = (activeSlide > 0) ? activeSlide-1 : images-1;
```
Comment (Gruszks) [inline js/slider.jquery.js]:
```
if () {
 CODE
}
```
Never in one line.

File: js/slider.jquery.js [hunk 17]
```diff
@@ -0,0 +1,99 @@
...
+                runWith: 0,
+                activeSlide: 0
+            }, options);
+
+
+            /*
+             * Wrapping exisiting listed slides into the plugin's layer.
+             */
+            slideshowDiv = $('<div/>').addClass('slideshow');
+            that = $(this);
```
Comment (Gruszks) [inline js/slider.jquery.js]: what is that? user better names of variables.

**Pull request closed without merging**